contracts/erc721/UserMintableERC721.sol

```diff
@@ -489,6 +491,15 @@ contract UserMintableERC721 is Initializable, ERC721, ERC721Enumerable, ERC721Me
   view
   {
 
+    require(
```

Comment:
It might be cheaper do not use the verify method here.

Because it is just one hash function call:

something like:

```javascript
      if (signingRoot  < singingRootProof[0]) {
        require(sha256(abi.encodePacked(signingRoot, singingRootProof[0]))==documentRoot)
      } else {
       require(sha256(abi.encodePacked(singingRootProof[0], signingRoot))==documentRoot)
}
```

What do you think?

Reply:
but I am not sure with requires in if conditions :)

Reply:
I think it complicates the code and even if it provides a gas saving the balance will be negative.

Reply:
the current approach is more consistent